packages/utils/src/isServiceWorkerSupported.ts

```diff
@@ -0,0 +1,45 @@
+import semver from 'semver';
+import { UAParser } from 'ua-parser-js';
+
+type TBrowsers =
+    | 'Chrome'
+    | 'Edge'
+    | 'Firefox'
+    | 'Opera'
+    | 'Safari'
+    | 'Opera Android'
+    | 'Mobile Safari'
+    // | 'Samsung Internet'
+    | 'Android Browser';
```

Comment:
I need to check the exact name of each browser using a simulator to make sure these values are correct.